generic.c

```diff
@@ -164,3 +164,33 @@ int createFile(const char * file, const char * buff, size_t size)
 	return SUCCESS;
 }
 
+/*
+ *returns a new pointer to a string array with one more unused index at the end 
+ *@param arr , a pointer to the string array
+ *			will be reallocated to have one more index at the end or NULL if error
+ *@param old_length , the current length of the array, the returned array will have old_length+1 memory locations
+ *@return 0 for success or ALLOC_FAIL if fail (memory will be freed in this case)
+ */
+int reallocArray(const char ***arr, int old_length)
```

Comment:
2 comments:
 - taking old_length just so you can allocate old_length + 1 is a bit odd. Maybe just take `new_length`.
 - if you make arr a void type, you can use it for any sort of array.